x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/policy_json_flyout.tsx

```diff
@@ -17,33 +18,95 @@ import {
   EuiSpacer,
   EuiText,
   EuiTitle,
+  EuiCallOut,
+  EuiLoadingSpinner,
 } from '@elastic/eui';
+
 import { SerializedPolicy } from '../../../../../common/types';
 
+import { useFormContext } from '../../../../shared_imports';
+
 interface Props {
-  readPolicy: () => SerializedPolicy;
   close: () => void;
   policyName: string;
 }
 
-export const PolicyJsonFlyout: React.FunctionComponent<Props> = ({
-  policyName,
-  readPolicy,
-  close,
-}) => {
-  const { phases } = readPolicy();
-  const json = JSON.stringify(
-    {
-      policy: {
-        phases,
-      },
-    },
-    null,
-    2
-  );
+export const PolicyJsonFlyout: React.FunctionComponent<Props> = ({ policyName, close }) => {
+  /**
+   * policy === undefined: we are checking validity
+   * policy === null: we have determined the policy is invalid
+   * policy === {@link SerializedPolicy} we have determined the policy is valid
+   */
+  const [policy, setPolicy] = useState<undefined | null | SerializedPolicy>(undefined);
+
+  const form = useFormContext();
+
+  useEffect(() => {
+    const subscription = form.subscribe(async (formUpdate) => {
```

Comment:
I will deprecate the `form.subscribe` in favor of the `useFormData`. Can we use it instead here?

Reply:
Sure! I think in this particular case, the solution was less verbose with the use of form.subscribe because form.validate lives in a different context. However, I can see that useFormData should be the primary mechanism for hooking into updates for most cases.